packages/superset-ui-legacy-preset-chart-big-number/src/BigNumber/BigNumber.tsx

```diff
@@ -185,32 +191,65 @@ class BigNumberVis extends React.PureComponent {
     );
   }
 
-  renderTrendline(maxHeight) {
+  renderTrendline(maxHeight: number) {
     const {
       width,
       trendLineData,
       mainColor,
       subheader,
-      renderTooltip,
       startYAxisAtZero,
+      formatNumber,
+      formatTime,
+      fromDatetime,
+      timeRangeFixed,
     } = this.props;
 
+    // Apply a fixed X range if a time range is specified.
+    //
+    // XYChart checks the existence of `domain` property decide whether to apply
+    // a domain or not, so it must not be `null` or `undefined`
+    const xScale: { type: string; domain?: number[] } = { type: 'timeUtc' };
+    const tooltipData = trendLineData?.sort(datum => datum.x);
```

Comment:
* This `.sort()` will mutate the original `props`, which violates `react` 's one-way data flow concept. Should make a copy first `trendlineData?.concat().sort(...)`
* It seems like `trendLineData` should already have been sorted in `transformProps`?